src/validation/validationSchema.ts

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+export default {
+    Common: {
+        Next: true,
+        End: true,
+        Comment: true,
+        InputPath: true,
+        OutputPath: true,
+        Type: true
+    },
+    Specific: {
+        Pass: {
+            Properties: {
+                Result: true,
+                ResultPath: true,
+                Parameters: true
+            }
+        },
+        Task: {
+            Properties: {
+                Resource: true,
+                Parameters: true,
+                ResultPath: true,
+                TimeoutSeconds: true,
+                HeartbeatSeconds: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retry'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catch'
+                }
+            }
+        },
+        Choice: {
+            Properties: {
+                Choices: {
+                    'Fn:ArrayOf': 'Choices'
+                },
+                Default: true
+            }
+        },
+        Wait: {
+            Properties: {
+                Seconds: true,
+                Timestamp: true,
+                SecondsPath: true,
+                TimestampPath: true
+            }
+        },
+        Succeed: {
+            Properties: {
+                Type: true
+            },
+            hasCommonFields: false
+        },
+        Fail: {
+            Properties: {
+                Cause: true,
+                Error: true,
+                Comment: true,
+                Type:  true
+            },
+            hasCommonFields: false
+        },
+        Parallel: {
+            Properties: {
+                Branches: true,
+                ResultPath: true,
+                Retry: {
+                    'Fn:ArrayOf': 'Retry'
+                },
+                Catch: {
+                    'Fn:ArrayOf': 'Catch'
+                }
+            }
+        },
+        Map: {
+            Properties: {
+                Iterator: true,
```

Comment:
Do we want to provide Iterator validation?

Reply:
Iterator is a nested state. The schema will be applied to every nested state.